### PR TITLE
Linting passes outta the box, updates routes to work, updates deps

### PR DESCRIPTION
**Pull request by AllenBW**
Just some basic housecleaning that has been long overdue

### looks the same
Screen Shot 2020-05-27 at 10 45 28 AM

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -1 +1,3 @@
-src/demoData/*
+node_modules/*
+static/*
+dist/*
```
Comment (Hyperkid123) [inline .eslintignore:3]: missing new line :boom:

File: src/Routes.js [hunk 2]
```diff
@@ -51,16 +48,20 @@ InsightsRoute.propTypes = {
...
-export const Routes = (props: Props) => {
-    const path = props.childProps.location.pathname;
+export const Routes = ({ childProps }) => {
+    const path = childProps.location.pathname;
 
     return (
         <Switch>
-            <InsightsRoute path={ paths.samplepage } component={ SamplePage } rootClass='samplepage'/>
-            <InsightsRoute path={ paths.rules } component={ Rules } rootClass='rules'/>
+            <InsightsRoute path={paths.samplepage} component={SamplePage} rootClass='samplepage' />
```
Comment (Hyperkid123) [inline src/Routes.js:56]: Do we still need the insights route? With the `React.lazy/Suspense` we can achieve the same thing but with official React support.

Comment (AllenBW) [reply src/Routes.js:56]: good point, i'll do that change here, no reason not to in this update

Comment (AllenBW) [reply src/Routes.js:56]: In this place, the `insightsroute` modifies attributes rather than fiddle with the loading...

File: src/Routes.js [hunk 3]
```diff
@@ -51,16 +48,20 @@ InsightsRoute.propTypes = {
...
-            { /* Finally, catch all unmatched routes */ }
-            <Route render={ () => some(paths, p => p === path) ? null : (<Redirect to={ paths.samplepage }/>) }/>
+            { /* Finally, catch all unmatched routes */}
+            <Route render={() => some(paths, p => p === path) ? null : (<Redirect to={paths.samplepage} />)} />
         </Switch>
     );
 };
+
+Routes.propTypes = {
+    childProps: PropTypes.object
```
Comment (Hyperkid123) [inline src/Routes.js:66]: I think we should kill this. And just use useLocation hook to get to the location.

File: src/Routes.js (hunk 2 above)
Comment (Hyperkid123) [reply src/Routes.js:56]: Is it really necessary to modify any attributes? The root class can be applied to the actual component.

Comment (AllenBW) [reply src/Routes.js:56]: Not looking to rewrite the starter app here, just made a new repo n thought I'd give back a lil bitta the work needed to get it going 🤷

Comment (Hyperkid123) [reply src/Routes.js:56]: Right I forgot

File: src/Routes.js (hunk 3 above)
Comment (AllenBW) [reply src/Routes.js:66]: OOOO i like that, ok lemme check its 🆗

Comment (AllenBW) [reply src/Routes.js]: yeah this is great sauce @Hyperkid123 updated <3

**Pull request merged**